src/Modules/RoundRankingModule/Templates/Components/RoundRankingRow.mt

```diff
@@ -0,0 +1,99 @@
+            <quad
+                    bgcolor="{{ Theme.UI_RoundRankingModule_Widget_AccentColor(pos) }}"
+                    size="0.7 {{ height }}"
+                    pos="0 0"
+            />
+            <quad
+                    bgcolor="{{ Theme.UI_AccentSecondary }}"
+                    size="{{ height }} {{ height }}"
+                    opacity="0.9"
+                    pos="0.7 0"
+            />
+            <label
+                    text="{{ pos }}"
+                    class="text-primary"
+                    pos="{{ 0.6 + height/2 }} {{ -height/2.0 + 0.2 }}"
+                    size="{{ height }} {{ height }}"
+                    valign="center"
+                    halign="center"
+                    textcolor="{{ Theme.Black }}"
+            />
+
+            <Panel
+                    width="{{ width-4 }}"
+                    height="4"
+                    className='{{ self ? "lr-body-highlight" : "lr-body-primary" }}'
+                    x="4.7"
+                    bgColor=""
+            >
+                <label
+                        text="{{ checkpoint.Player.NickName }}"
+                        class="text-primary"
+                        textsize="0.5"
+                        pos="1 {{ -height/2.0 + 0.2 }}"
+                        valign="center"
+                        size="{{ width-20 }} {{ height }}"
+                />
+                <label
+                        text="{{ checkpoint.IsDNF ? &quot;DNF&quot; : (checkpoint.TimeDifference!=null ? checkpoint.TimeDifference.ToString().TrimStart('0') : checkpoint.Time.ToString().TrimStart('0') ) }}"
+                        textprefix='{{ checkpoint.TimeDifference!=null&amp;&amp;!checkpoint.IsDNF ? "+" : "" }}'
+                        class="text-primary"
+                        textsize="0.5"
+                        pos="{{ width-4-(height * 1.25)-1.0 }} {{ -height/2.0 + 0.2 }}"
+                        valign="center"
+                        halign="right"
+                        size="9 {{ height }}"
+                        opacity="{{ checkpoint.TimeDifference==null&amp;&amp;!checkpoint.IsDNF ? 1.0 : 0.75 }}"
+                />
+                <frame>
+                    <!-- CHECKPOINT ID & GAINED POINTS BOX -->
+                    <quad
+                            size="{{ height * 1.25 }} {{ height }}"
+                            pos="{{ width-4 }}"
+                            class='lr-body-highlight'
+                            bgcolor="{{ checkpoint.AccentColor != null ? checkpoint.AccentColor : Theme.UI_RoundRankingModule_Widget_RowBgHighlight }}"
+                            halign="right"
+                    />
+                    <label
+                            if="checkpoint.GainedPoints == 0"
+                            class="text-primary"
+                            text='{{ checkpoint.IsDNF ? Icons.FlagO : (checkpoint.IsFinish ? Icons.FlagCheckered : checkpoint.CheckpointId + 1) }}'
```

Comment:
Maybe some sort of utility method would be better here than nesting ternary statements.

Actually would be nice if there the template engine had a way to define C# methods somewhere in the component that can be reused and make the code cleaner.